ui/src/app/components/workflows/workflows-home/workflows-home.component.ts

```diff
@@ -78,9 +102,39 @@ export class WorkflowsHomeComponent implements OnInit, OnDestroy {
     this.router.navigate([absoluteRoutes.SHOW_WORKFLOW, id]);
   }
 
+  onClarityDgRefresh(state: ClrDatagridStateInterface) {
+    if (!this.ignoreRefresh) {
+      this.sort = state.sort ? new SortAttributesModel(state.sort.by as string, state.sort.reverse ? -1 : 1) : undefined;
+      this.store.dispatch(new SetWorkflowsSort(this.sort));
```

Comment:
Why is it necessary to dispatch an event? What's the difference to `runs.ts`, where it's not needed?

Reply:
Yes, it is necessary, RunsComponents fetches the data from the server, while here the idea was to filter/sort only from the front-end, so to achieve this is by dispatching/keeping the current sort and/or filter state(s) to the store.

Reply:
Yes, Senele is right.